optimum/exporters/onnx/utils.py

```diff
@@ -323,6 +323,7 @@ def get_stable_diffusion_models_for_export(
         text_encoder_config_constructor = TasksManager.get_exporter_config_constructor(
             model=pipeline.text_encoder,
             exporter="onnx",
+            library_name="diffusers",
```

Comment:
the `library_name` of the clip text model should be transformers: https://huggingface.co/stabilityai/stable-diffusion-2-1/blob/main/model_index.json. In the pipeline, only unet and vae's implementation classes are from diffusers.

Reply:
@JingyaHuang `clip-text-model` can not be exported through `main_export` (see https://github.com/huggingface/optimum/blob/45c1c095e6bbf498ea2e10d3d8449858e037b76c/optimum/exporters/tasks.py#L1056), and is currently diffusers-specific: https://github.com/fxmarty/optimum/blob/22286de80b1da7135566bea5fafa2c04145b8961/optimum/exporters/tasks.py#L296-L303

Reply:
@JingyaHuang `_UNSUPPORTED_CLI_MODEL_TYPE` was introduced by @echarlaix for the stable diffusion export, where some entries (unet, etc.) were added in `_SUPPORTED_MODEL_TYPES` but could nevertheless not be exported with `main_export`.

It is not particularly confusing, given that as I linked (https://github.com/fxmarty/optimum/blob/22286de80b1da7135566bea5fafa2c04145b8961/optimum/exporters/tasks.py#L296-L303) the `clip-text-model` is specific to the export of diffusers models, just like it was before.

This [does not matter](https://github.com/huggingface/optimum/blob/da6f9e2d9bc57c4a337ac3a30a831f295325a199/optimum/exporters/onnx/__main__.py#L390) for the model loading since `get_model_from_task` `get_model_class_for_task` just use the top level `library_name` to load the full pipeline. This is just a matter of loading the ONNX config.

If we want to change that, I suggest to do so in an other PR.